mobilizon_reshare/cli/commands/publish/main.py

```diff
@@ -1,10 +1,39 @@
-from mobilizon_reshare.main.publish import publish
+import logging
 
+from mobilizon_reshare.main.publish import (
+    select_and_publish,
+    publish_event,
+    publish_publications,
+)
+from mobilizon_reshare.publishers import get_active_publishers
+from mobilizon_reshare.storage.query.read import get_mobilizon_event, get_publication
 
-async def publish_command():
+logger = logging.getLogger(__name__)
+
+
+async def publish_command(event, publication, platform, list_platforms):
     """
-    STUB
-    :return:
+    If no arguments are provided, select an event with the current configured strategy
+    and publish it to all active platforms.
+
+    - If list_platforms is True print all platforms and exit
+    - Otherwise:
+       + If publication is not None, publish it.
+       + Otherwise if event is not None publish it.
     """
-    reports = await publish()
-    return 0 if reports and reports.successful else 1
+    if publication and event:
+        logger.error("-P/--publication and -E/--event are incompatible!")
+        return 1
+
+    if list_platforms:
+        for platform in get_active_publishers():
+            print(f"\t{platform}")
+        return 0
+
+    if publication:
```

Comment:
these should be two separate commands.

Keeping them together adds complexity and reduces clarity. Also the CLI handling different operations through arguments is less than ideal.

Reply:
So, the rationale behind this PR is the following:

We began with the `start` command which did all kind of magic. Exactly the minimum set of operations required for `mobilizon-reshare`'s day-to-day, but all bundled within the same command.
Then we divided `start` in `pull` and `publish` because they were lower level operations. They are lower level because you can implement `start` by composing `pull` and `publish` but not viceversa.

My point being that we should strive to try and expose a set of primitives which can then be composed by the user. We can implement `retry` in terms of `publish` but not viceversa. I'm not arguing for the removal of higher level commands like `start` or `retry`, I just feel we should allow the user for as much control as possible.

> Also the CLI handling different operations through arguments is less than ideal.
I don't see how they are different operations, the very same act of publishing an event is modeled by a `Publication`. If we define `publish` as the operation that receives a `list[Publication]` and tries to publish them,  we can implement `select_and_publish` in terms of `publish` but not viceversa.

By this logic it would make more sense to move `select_and_publish` to another command, but it could result in bad UX.

Reply:
I think you focus too much on the abstract relationship between Event and Publication and not enough on how a user would use the tool. Even though it can only be speculative reasoning, it's easy to imagine that the following 4 operations happen in very different contexts and moments:
* publishing a whole event
* publishing a single publication
* retry a whole event
* retry a single publication


It's also legitimate to think that users should be guided to discover the difference between an event and a publication through the CLI if they don't understand well the difference. Putting them as a flag makes them look like something taking a space very different in the logic of mo-re than what they actually are. Big things should look big, small things should look small. Here we are doing the opposite.

Reply:
I moved everything under retry command to try and remove the duplication of logic. It probably could help some refactoring of the modules structure.